packages/os/bootstrap-containers@.service

```diff
@@ -27,3 +28,6 @@ ExecStartPost=/usr/bin/bootstrap-containers mark-bootstrap \
     --mode '${CTR_MODE}'
 RemainAfterExit=true
 StandardError=journal+console
+
+[Install]
+RequiredBy=reboot-to-reconcile.service
```

Comment:
This will be overridden by a drop-in configuration:

https://github.com/bottlerocket-os/bottlerocket/blob/037d0e2798a7e8d4a12d0fb35d3c266db3d74f25/sources/api/bootstrap-containers/src/main.rs#L376

Regardless of ^, bootstrap containers shouldn't be `RequiredBy=reboot-to-reconcile.service`. Non-essential bootstrap containers don't block the boot sequence when they fail. By making all bootstrap containers `RequiredBy=reboot-to-reconcile.service`, failed non-essential bootstrap containers will indirectly block the boot sequence since `reboot-to-reconcile.service` is `RequiredBy=configured.target`.

Reply:
The right way to accomplish this might be to order `reboot-to-reconcile.service` via `activate-multi-user.service`:
```
diff --git a/packages/release/activate-multi-user.service b/packages/release/activate-multi-user.service
index 964301e9..bdb6f475 100644
--- a/packages/release/activate-multi-user.service
+++ b/packages/release/activate-multi-user.service
@@ -1,7 +1,7 @@
 [Unit]
 Description=Isolates multi-user.target
-After=configured.target
-Requires=configured.target
+After=configured.target reboot-to-reconcile.service
+Requires=configured.target reboot-to-reconcile.service
 
 [Service]
 Type=oneshot
```

These "activate" units run in a sort of weird limbo state - they're pulled in by the target, but only run after that target has been reached. We haven't had to do anything besides the target switch in these states before, but conceptually this is when we want the reconciliation reboot to happen.

Reply:
I don't think the drop-in created by bootstrap-containers is going to override the `RequiredBy` in the template, it should rather add an extra dependency. `systemd-analyze dot` seems to confirm this. Nonetheless, you're entirely right I missed to consider non-essential bootstrap containers. It goes to show the way the ordering was achieved here is rather clumsy.

I like Ben's suggestion. It takes care of the required distinction of essential/non-essential bootstrap containers and also appears semantically right: The host should reboot once it has been configured.

Reply:
I forgot to mention that you'd also need to move `reboot-to-reconcile.service` to be `After=configured.target` instead of before, so it doesn't get started earlier than expected in the `configured.target` phase.